src/code/z_en_item00.c

```diff
@@ -881,59 +872,50 @@ EnItem00* Item_DropCollectible(GlobalContext* globalCtx, Vec3f* spawnPos, u32 pa
         }
     } else {
         if (param8000 == 0) {
-            params = func_800A7650(params & 0xFF);
+            params = func_800A7650(newParamFF);
         }
-        if (params != (u32)ITEM00_NO_DROP) {
-            spawnedActor = (EnItem00*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00, spawnPos->x,
-                                                  spawnPos->y, spawnPos->z, 0, 0, 0, params | param8000 | param7F00);
+        if ((s32)params != ITEM00_NO_DROP) {
+            spawnedActor = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00, spawnPos->x, spawnPos->y,
+                                       spawnPos->z, 0, 0, 0, (s32)params | param8000 | param7F00);
```

Comment:
These casts still needed?

Reply:
the conditional no, the or does it need for regalloc reasons.